### PR TITLE
Replace setup-scala with setup-java

### PR DESCRIPTION
**Pull request by mijicd**
### Summary

- replaces setup-scala with setup-java
- sets Java 17 as default

### CODE REVIEW

**Review by guizmaii — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -99,15 +119,17 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: ['11', '17']
+        java: ['8', '11']
```
Comment (sideeffffect) [inline .github/workflows/ci.yml:122]: The purpose of the task `testJvms` is to test on other JVM versions _besides the base one_, so it should stay `['8', '11']`.

**Review by sideeffffect — COMMENTED** [review]
IMHO we should keep JVM version 8 as _the base version_ and use it everywhere. Other JVM versions, like 11 or 17 should be then tested only in the `testJvms`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mijicd) [reply .github/workflows/ci.yml:122]: It should not, as the base one is 17.

**Comment by mijicd** [comment]
> The motivation is that this will make sure that all things work on the widest range of JVM versions -- this is one of Scala's biggest strengths, it is a powerful language that you can run even on an "old" JVM 8.

If I'm not mistaken, that's the whole purpose behind the `testJvms`.

**Review by adamgfraser — APPROVED** [review]

**Comment by mijicd** [comment]
@adamgfraser Could you please sync the `series/2.x` branch?

**Comment by adamgfraser** [comment]
@mijicd Will do!

**Comment by sideeffffect** [comment]
But how about the other actions, like `lint`, `compile`, `publishLocal` or `website`? Shouldn't we primarily test them on 8 (instead of 17)? :thinking:

**Comment by mijicd** [comment]
`publishLocal` and `publish` are still running on 8, due to the downward dependency. Regarding the other ones, I don't see that we're risking anything with migration to 17.

**Comment by sideeffffect** [comment]
fair enough :+1:

**Pull request merged**